### PR TITLE
Fixes a bug related to CR2 RAW

### PR DESCRIPTION
**Pull request by miaoqi**
* reformats and cleans code in updated source files
* changes the camera model from “5d” to “eos 5d max ii” on the second line of “data/camera/Canon_5D_380_780_5”
* updates “loadCameraSpst ()” function of Idt class to determine if the given spectral sensitivity data is external
* uses “uchar” to “ushort” for image “data” array type-conversion after calling “dcraw_process ()” for further operations that are based on floating-point values
* adds additional condition in “prepareIDT ()” and “prepareWB()” to make sure the spectral sensivity data is eternal or already included
* solve #24

### CODE REVIEW

File: data/camera/Canon_5D_380_780_5 [hunk 1]
```diff
@@ -1,5 +1,5 @@
 canon
-5d
+eos 5d mark ii
```
Comment (lchen198) [inline data/camera/Canon_5D_380_780_5:2]: Will you consider changing this file into a json or other self-describing data types

File: lib/mathOps.h [hunk 2]
```diff
@@ -271,7 +271,8 @@ void minusVector(vector<T>& vct, T sub)
 }
 
 template <typename T>
-vector<T> mulVectorElement(const vector<T>& vct1, const vector<T>& vct2)
```
Comment (lchen198) [inline lib/mathOps.h:274]: Do you have test for these low-level functions? I guess you must have some constraint that you can not use alternative well-tested libraries

File: src/acesrender.cpp [hunk 3]
```diff
@@ -156,8 +157,9 @@ int AcesRender::readCameraSenPath( const char * cameraSenPath,
                 string fn( *file );
 
                 readC = _idt->loadCameraSpst( fn,
-                                             static_cast <const char *> (P.make),
-                                             static_cast <const char *> (P.model) );
+                                              static_cast <const char *> (P.make),
```
Comment (lchen198) [inline src/acesrender.cpp:160]: Do you have a code formating tool that can be used by everyone on the team?

File: lib/mathOps.h (hunk 2 above)
Comment (miaoqi) [reply lib/mathOps.h:274]: That is a good point. I have though about using eigen library.

File: src/acesrender.cpp (hunk 3 above)
Comment (miaoqi) [reply src/acesrender.cpp:160]: Can you recommend some tools? Thanks!

File: lib/mathOps.h (hunk 2 above)
Comment (miaoqi) [reply lib/mathOps.h:274]: Are you aware of any good math libraries that are available on various platforms including these old Linux systems?

**Pull request merged**